### PR TITLE
Update to 0.5.0 release

### PR DESCRIPTION
**Pull request by leshikus**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -33,7 +31,7 @@ outputs:
     script: build-hdk.sh
     build:
       string: {{ py_build_string_ext }}
-      skip: True  # [py not in (38, 39, 310)]
+      skip: True  # [py not in (38, 39, 310, 311)]
```
Comment (Garra1980) [inline recipe/meta.yaml:34]: let's leave this as is for now

File: recipe/meta.yaml [hunk 2]
```diff
@@ -64,18 +62,18 @@ outputs:
         - make
         - maven
         - ninja
+        - tbb-devel
```
Comment (Garra1980) [inline recipe/meta.yaml:65]: you sure this is needed?

Comment (Garra1980) [reply recipe/meta.yaml:65]: oh, it must have been replaced from other section...

File: recipe/meta.yaml (hunk 1 above)
Comment (leshikus) [reply recipe/meta.yaml:34]: this is draft, I'm testing it

File: .ci_support/linux_64_c_compiler_version11cuda_compiler_versionNonecxx_compiler_version11numpy1.21python3.9.____cpython.yaml [hunk 3]
```diff
@@ -23,7 +23,7 @@ docker_image:
 glog:
 - '0.6'
 numpy:
-- '1.20'
+- '1.21'
```
Comment (Garra1980) [inline .ci_support/linux_64_c_compiler_version11cuda_compiler_versionNonecxx_compiler_version11numpy1.21python3.9.____cpython.yaml:26]: Is this changed by conda smithy itself?

Comment (leshikus) [reply .ci_support/linux_64_c_compiler_version11cuda_compiler_versionNonecxx_compiler_version11numpy1.21python3.9.____cpython.yaml:26]: yes, these changes come from conda-smithy

File: recipe/meta.yaml (hunk 1 above)
Comment (leshikus) [reply recipe/meta.yaml]: code outdated

**Review by Garra1980 — APPROVED** [review]
I'd say we're good to go

**Pull request merged**